train.py

```diff
@@ -198,13 +200,16 @@ def train(self):
 
     best_valid_loss = 1.e6
     for epoch in range(self.startEpoch, self.params.max_epochs):
+      logging.info("Epoch: %d" % epoch)
       if dist.is_initialized():
         self.train_sampler.set_epoch(epoch)
 #        self.valid_sampler.set_epoch(epoch)
 
       start = time.time()
       tr_time, data_time, train_logs = self.train_one_epoch()
       valid_time, valid_logs = self.validate_one_epoch()
+      self.inference_one_epoch()
```

Comment:
I think the cleanest way to deal with the wandb step vs epoch issue would be to return the desired metrics/images as dicts from the `self.validate_one_epoch()` and `self.inference_one_epoch()` functions. Then these could be merged and passed to `wandb.log(...)` in a single call. This would require some refactoring of the `inference.py` script as well.

Reply:
sounds good